dev/com.ibm.ws.microprofile.faulttolerance_fat_tck/fat/src/com/ibm/ws/microprofile/tck/Utils.java

```diff
@@ -0,0 +1,299 @@
+	 */
+	public static Map<String, String> resolveJarPaths(List<String> jars, LibertyServer server) {
+		HashMap<String, String> result = new HashMap<String, String>(jars.size());
+		for (Iterator<String> iterator = jars.iterator(); iterator.hasNext();) {
+			String jarName = iterator.next();
+			String jarPath = resolveJarPath(jarName, server);
+			result.put(jarName, jarPath);
+		}
+		return result;
+	}
+
+	/**
+	 * Return a full path for a jar file name substr. This function enables the
+	 * Liberty build version (which is often included in jar names) to increment and
+	 * the FAT bucket to find the jar under the new version.
+	 *
+	 * @param jarName
+	 * @param server
+	 * @return
+	 */
+	public static String resolveJarPath(String jarName, LibertyServer server) {
+		String wlp = server.getInstallRoot();
+		String jarPath = genericResolveJarPath(jarName, wlp);
+		return jarPath;
+	}
+
+	/**
+	 * This is more easily unit testable and reusable version of the internals of
+	 * resolveJarPath
+	 *
+	 * @param jarName
+	 * @param wlpPathName
+	 * @return the path to the jar
+	 */
+	public static String genericResolveJarPath(String jarName, String wlpPathName) {
+		String dev = wlpPathName + "/dev/";
+		String api = dev + "api/";
+		String spi = dev + "spi/";
+		String apiStable = api + "stable/";
+		String lib = wlpPathName + "/lib/";
+
+		ArrayList<String> places = new ArrayList<String>();
+		places.add(apiStable);
+		places.add(lib);
+
+		String jarPath = null;
+		for (Iterator<String> iterator = places.iterator(); iterator.hasNext();) {
+			String dir = iterator.next();
+			log("JAR: dir=" + dir);
+			jarPath = jarPathInDir(jarName, dir);
+			if (jarPath != null) {
+				log("JAR: dir match=" + dir + jarPath);
+				jarPath = dir + jarPath;
+				break;
+			}
+		}
+		return jarPath;
+	}
+
+	/**
+	 * Looks for a path in a directory from a sub set of the filename
+	 *
+	 * @param jarNameFragment
+	 * @param dir
+	 * @return
+	 */
+	public static String jarPathInDir(String jarNameFragment, String dir) {
+		String result = null;
+		File dirFileObj = new File(dir);
+		String[] files = dirFileObj.list();
+		log("looking for jar " + jarNameFragment + " in dir " + dir);
+
+		// Looking for (for example):
+		// /libertyGit/open-liberty/dev/build.image/wlp/lib
+		// $ find . -name "*tolerance*.jar"
+		// ./com.ibm.ws.microprofile.faulttolerance.cdi_1.0.18.jar
+		// ./com.ibm.ws.microprofile.faulttolerance_1.0.18.jar
+		// ./com.ibm.ws.microprofile.faulttolerance.spi_1.0.18.jar
+		for (int i = 0; i < files.length; i++) {
+			if (files[i].contains(jarNameFragment)) {
+				result = files[i];
+				log("dir " + dir + " contains " + jarNameFragment + " as " + result);
+				// Do we have a match ignoring version numbers
+				String r = result.replaceAll(".jar", "");
+				r = r.replaceAll("[0-9]", "");
+				r = r.replaceAll("\\.", "");
+				r = r.replaceAll("\\_", "");
+				log("r testing " + r);
+				if (r.equals(jarNameFragment.replaceAll("\\.", ""))) {
```

Comment:
This code could do with a few more comments on how the matching works. e.g. why remove the dots from both strings?